### ISSUE TITLE
[style-prop-object] False error: Style prop value must be an object

### ISSUE THREAD

**Issue by tleunen** [yannickcr/eslint-plugin-react#820]
The rule is throwing in my code with this example:

```jsx
React.createElement(MyCustomElem, {
    [style]: true
}, `My custom Elem`)
```

if I rename the variable `style` with something else, everything is fine.

```
Cannot read property 'name' of undefined
TypeError: Cannot read property 'name' of undefined
    at /Users/tommy.leunen/dev/pppp/node_modules/eslint-plugin-react/lib/rules/style-prop-object.js:51:34
    at Array.find (native)
```
